packages/flutter/lib/src/material/list_tile.dart

```diff
@@ -1278,187 +1306,166 @@ class _RenderListTile extends RenderBox with SlottedContainerRenderObjectMixin<_
-    final BaselineOffset offset = BaselineOffset(title!.getDistanceToActualBaseline(baseline))
+    final BoxParentData parentData = title.parentData! as BoxParentData;
+    final BaselineOffset offset = BaselineOffset(title.getDistanceToActualBaseline(baseline))
                                 + parentData.offset.dy;
     return offset.offset;
   }
 
-  static double? _boxBaseline(RenderBox box, TextBaseline baseline) {
-    return box.getDistanceToBaseline(baseline);
-  }
-
-  static Size _layoutBox(RenderBox? box, BoxConstraints constraints) {
-    if (box == null) {
-      return Size.zero;
-    }
-    box.layout(constraints, parentUsesSize: true);
-    return box.size;
-  }
+  BoxConstraints get maxIconHeightConstraint => BoxConstraints(
+    // One-line trailing and leading widget heights do not follow
+    // Material specifications, but this sizing is required to adhere
+    // to accessibility requirements for smallest tappable widget.
+    // Two- and three-line trailing widget heights are constrained
+    // properly according to the Material spec.
+    maxHeight: (isDense ? 48.0 : 56.0) + visualDensity.baseSizeAdjustment.dy,
+  );
 
   static void _positionBox(RenderBox box, Offset offset) {
     final BoxParentData parentData = box.parentData! as BoxParentData;
     parentData.offset = offset;
   }
 
-  @override
-  Size computeDryLayout(BoxConstraints constraints) {
-    assert(debugCannotComputeDryLayout(
-      reason: 'Layout requires baseline metrics, which are only available after a full layout.',
-    ));
-    return Size.zero;
-  }
-
+  // Implements _RenderListTile's layout algorithm. If `positionChild` is not null,
+  // it will be called on each child with that child's layout offset.
+  //
   // All of the dimensions below were taken from the Material Design spec:
   // https://material.io/design/components/lists.html#specs
-  @override
-  void performLayout() {
-    final BoxConstraints constraints = this.constraints;
-    final bool hasLeading = leading != null;
-    final bool hasSubtitle = subtitle != null;
-    final bool hasTrailing = trailing != null;
-    final bool isTwoLine = !isThreeLine && hasSubtitle;
-    final bool isOneLine = !isThreeLine && !hasSubtitle;
-    final Offset densityAdjustment = visualDensity.baseSizeAdjustment;
-
-    final BoxConstraints maxIconHeightConstraint = BoxConstraints(
-      // One-line trailing and leading widget heights do not follow
-      // Material specifications, but this sizing is required to adhere
-      // to accessibility requirements for smallest tappable widget.
-      // Two- and three-line trailing widget heights are constrained
-      // properly according to the Material spec.
-      maxHeight: (isDense ? 48.0 : 56.0) + densityAdjustment.dy,
-    );
+  _Sizes _computeSizes(
+    ChildBaselineGetter getBaseline,
+    ChildLayouter getSize,
+    BoxConstraints constraints,
+    { _PositionChild? positionChild, }
+  ) {
```

Comment:
nit: our formatting style here (I think) is:

```suggestion
    BoxConstraints constraints, {
    _PositionChild? positionChild,
  }) {
```